src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -209,7 +216,15 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                     // Only set our external startup if we're not suppressing host initialization
                     // as we don't want to load user assemblies otherwise.
                     var locator = context.Properties.GetAndRemove<ScriptStartupTypeLocator>(StartupTypeLocatorKey);
-                    webJobsBuilder.UseExternalStartup(locator, webJobsBuilderContext, loggerFactory);
+
+                    try
+                    {
+                        webJobsBuilder.UseExternalStartup(locator, webJobsBuilderContext, loggerFactory);
```

Comment:
Question for my learning: What if the external startup class/method has some infinite loop or similar thing (Similar to that recent CRI you helped with)? I looked at the `UseExternalStartup` method and do not see any kind of timeout mechanism we implemented there to throw an exception if the customer startup code did not finish execution in X number of seconds.

Reply:
I don't think we explicitly handle this ourselves anywhere. It's worth investigating whether we can add something here (in a follow-up).

Reply:
looks like i've already filed that here: https://github.com/Azure/azure-functions-host/issues/8703